### ISSUE TITLE
[v2 FEATURE]: saving probability or the label of the predicted class as output for multiclass classification?

### ISSUE THREAD

**Issue by shihchengli** [chemprop/chemprop#979]
**Is your feature request related to a problem? Please describe.**
For binary classification, the output saved in the .csv file is a probability. However, for multiclass classification, the prediction is saved as probability for each class in the `train` CLI and the class label in the `predict` CLI. For consistency, they should be saved the same way in both the `train` and `predict` CLI. Additionally, providing the predicted probability (after sofmax) for each class is more informative (as users can calculate the class with the largest probability as the predicted class), but I can also see that providing the predicted class would be more straightforward. We can consider providing both by default, or one of them as the default with the other accessible via a flag in the CLI.

**Discussion**
1. How do we save the results for multiclass classification tasks?
2. If we are going to saved the probability as output in .csv file. Should we saved it as a list of floats (e.g., `[0.15, 0.8, 0.05]` for `n_classes = 3`) as the entry of the prediction result?

**Comment by davidegraff** [comment]
IMO I think binary should output a probability of class 1 at test time (i.e., `y_hat.sigmoid()`) rather than a raw logit. I think we would probably want to output probabilities as well for multiclass (i.e,. `y_hat.softmax(dim=-1)`). IDK what the best serialization format would be, but a simple quoted list should do the trick (`"0.15,0.8,0.05"`). If you actually output a python-style list, then you would need to use `ast.literal_eval()`, which is a whole complication that's really not worth it.

**Comment by shihchengli** [comment]
Hi @davidegraff, I used the wrong word here. I did mean probability instead of logits. Thanks for the clarification. I have changed the description and title of this issue. I agree that we should save the probability of class 1 for binary classification, which is what it’s currently doing. I would love to use the format you suggested for saving the multiclass classification results. BTW, we can also follow the approach we will change here in atom/bond property prediction (#475).
